### PR TITLE
Issue #13097: Split misc xdocs into separate pages

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #13097

### CODE REVIEW

**Comment by stoyanK7** [comment]
Github, generate site

**Review by romani — APPROVED** [review]
ok to merge

File: src/xdocs/checks.xml [hunk 1]
```diff
@@ -971,11 +990,15 @@
               Checks that type names conform to a specified pattern.</td>
           </tr>
           <tr>
-            <td><a href="config_misc.html#UncommentedMain">UncommentedMain</a></td>
+            <td><a href="checks/misc/uncommentedmain.html#UncommentedMain">UncommentedMain</a></td>
             <td>Detects uncommented <code>main</code> methods.</td>
```
Comment (nrmancuso) [inline src/xdocs/checks.xml:994]: Somehow we lost "`main` methods" from UncommentedMain description:  https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/23a9ce7_2023105251/checks/misc/index.html

See live site: https://checkstyle.sourceforge.io/checks.html#:~:text=Detects%20uncommented%20main%20method

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

File: src/xdocs/checks.xml (hunk 1 above)
Comment (stoyanK7) [reply src/xdocs/checks.xml:994]: @nrmancuso Yes, it's an issue with other checks as well. Present in https://github.com/checkstyle/checkstyle/pull/13264 too.

Comment (stoyanK7) [reply src/xdocs/checks.xml:994]: @romani 
Screenshot from 2023-06-21 16-21-48
Screenshot from 2023-06-21 16-22-09

Comment (stoyanK7) [reply src/xdocs/checks.xml:994]: Opened PR with test at https://github.com/checkstyle/checkstyle/pull/13285

Comment (stoyanK7) [reply src/xdocs/checks.xml:1057]: Fixed descriptions in table.

**Review by nrmancuso — APPROVED** [review]
@stoyanK7 please resolve conflicts, I am good to merge once CI is happy

**Comment by stoyanK7** [comment]
@nrmancuso Done.

**Pull request merged**